packages/react-ui-core/src/Gmap/Marker.js

```diff
@@ -0,0 +1,63 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import { MapContext } from './MapContext'
+
+const EVENTS = {
+  onClick: 'click',
+  onDoubleClick: 'dblclick',
+  onDragEnd: 'dragend',
+  onMouseDown: 'mousedown',
+  onMouseOut: 'mouseout',
+  onMouseOver: 'mouseover',
+  onMouseUp: 'mouseup',
+  onRecenter: 'recenter',
+}
+
+class Marker extends PureComponent {
+  componentWillUnmount() {
+    if (this.marker) this.marker.setMap(null)
+  }
+
+  setupMarker(map) {
+    if (map) {
+      this.marker = new window.google.maps.Marker({
+        ...this.props,
+        map,
+      })
+
+      this.setupEvents()
+    }
+
+    return null
+  }
+
+  setupEvents() {
+    Object.keys(EVENTS).forEach(name => {
+      if (this.props[name]) {
+        this.marker.addListener(EVENTS[name], this.handleEvent(name))
+      }
+    })
+  }
+
+  handleEvent(name) {
+    return event => {
+      this.props[name](event, this.props, this.marker)
+    }
+  }
+
+  render() {
+    return (
+      <MapContext.Consumer>
+        { context => this.setupMarker(context.map)}
+      </MapContext.Consumer>
+    )
+  }
+}
+
+Marker.propTypes = {}
+
+Object.keys(EVENTS).forEach(name => {
```

Comment:
is it worthwhile to memoize `Object.keys(EVENTS)` since you are using on line 35 as well?